src/figuresClasses.ts

```diff
@@ -1,19 +1,80 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 export interface Figure {
-
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      this.a <= 0
+      || this.b <= 0
+      || this.c <= 0
+      || this.a + this.b <= this.c
+      || this.b + this.c <= this.a
+      || this.a + this.c <= this.b
+    ) {
+      throw new Error('This triangle cannot exist');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const halfOfPerimetr = (this.a + this.b + this.c) / 2;
+    const triangleArea = Math.sqrt(halfOfPerimetr * (halfOfPerimetr - this.a)
+      * (halfOfPerimetr - this.b) * (halfOfPerimetr - this.c));
 
+    return Math.round(triangleArea * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('This circle cannot exist');
+    }
+  }
+
+  getArea(): number {
+    const circleArea = Math.PI * this.radius ** 2;
 
+    return Math.floor(circleArea * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: Shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('This rectangle cannot exist');
```

Comment:
Checklist p3

How user should know what exactly was wrong and what data are valid?